src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -282,25 +282,39 @@ void GcsActorManager::HandleCreateActor(rpc::CreateActorRequest request,
                 << ", actor id = " << actor_id;
   Status status = CreateActor(
       request,
-      [reply, send_reply_callback, actor_id](const std::shared_ptr<gcs::GcsActor> &actor,
-                                             const rpc::PushTaskReply &task_reply,
-                                             bool creation_cancelled) {
-        if (creation_cancelled) {
+      [reply, send_reply_callback, actor_id](
+          const std::shared_ptr<gcs::GcsActor> &actor,
+          const rpc::PushTaskReply &creation_task_reply,
+          const Status &creation_task_status) {
+        if (creation_task_status.IsCreationTaskError()) {
+          RAY_LOG(INFO) << "Failed to create actor due to init method failure, job id = "
+                        << actor_id.JobId() << ", actor id = " << actor_id;
+          // Actor creation task failed but we do have the actor worker address.
+          reply->mutable_actor_address()->CopyFrom(actor->GetAddress());
+          // We still keep track this despite of task failure since we don't destroy the
+          // actors right away but rely on actors being destroyed later when the worker
+          // exits. So we will need to inform the GCS on the borrowed refs as if the
+          // creation task was successful .
+          reply->mutable_borrowed_refs()->CopyFrom(creation_task_reply.borrowed_refs());
+          GCS_RPC_SEND_REPLY(send_reply_callback, reply, creation_task_status);
```

Comment:
```suggestion
          // Notify the owner so that owner can correctly report failure events for actor creation
          GCS_RPC_SEND_REPLY(send_reply_callback, reply, creation_task_status);
```